tests/monitoring/monitoring.go

```diff
@@ -630,6 +634,92 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
+				Patches: []v1.CustomizeComponentsPatch{
+					{
+						ResourceName: virtHandler.deploymentName,
+						ResourceType: "DaemonSet",
+						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
+						Type:         v1.StrategicMergePatchType,
+					},
+				},
+			}
+
+			Eventually(func() error {
+				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+				return err
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			Eventually(func() int {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return int(vh.Status.NumberAvailable + vh.Status.NumberUnavailable)
+			}, 90*time.Second, 5*time.Second).Should(Equal(0))
+
+			return kv
+		}
+
+		setEmulation := func(newStatus bool) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = newStatus
+
+			Eventually(func() error {
+				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+				return err
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+			Eventually(func() bool {
+				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			}, 90*time.Second, 5*time.Second).Should(Equal(newStatus))
+		}
+
+		restoreVirtHandler := func(kv *v1.KubeVirt) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
+
+			Eventually(func() error {
+				kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+				return err
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		}
+
+		It("KubeVirtNoAvailableNodesToRunVMs should be triggered when there are no available nodes in the cluster to run VMs", func() {
+			By("Scaling down virt-handler")
+			kv := disableVirtHandler()
+
+			By("Verifying KubeVirtNoAvailableNodesToRunVMs alert exists if emulation is disabled")
+			setEmulation(false)
+			verifyAlertExistWithCustomTime("KubeVirtNoAvailableNodesToRunVMs", 10*time.Minute)
+
+			By("Restoring virt-handler")
+			restoreVirtHandler(kv)
```

Comment:
You can drop this. This will be done by test suite.

Reply:
if we remove this, the alert will still be firing after the specific test, which may cause trouble for other tests in the future. The test suite will only restore in the end of all tests, right?

Reply:
The test suite restore it after each test, see https://github.com/kubevirt/kubevirt/blob/main/tests/tests_suite_test.go#L100

Reply:
I need the `restoreVirtHandler(kv)` to be able to `waitUntilAlertDoesNotExist("KubeVirtNoAvailableNodesToRunVMs")`. It was in the `AfterEach`, but I moved it to the test.